### DISCUSSION TITLE
WebDriverWait behaves differently with Firefox?

### DISCUSSION THREAD

**Discussion by ildarius** [php-webdriver/php-webdriver#894] [Q&A]
I recently switched from Chrome to Firefox standalone server and now the following 2 lines of code are returning an error

```
$wait = new WebDriverWait($driver, 30);
$wait->until(WebDriverExpectedCondition::visibilityOfElementLocated(WebDriverBy::xpath('//select/option[@value="50"]')));
```

the error is:

```
Fatal error: Uncaught Facebook\WebDriver\Exception\TimeoutException in WebDriverWait.php:71
Stack trace:
#0 : Facebook\WebDriver\WebDriverWait->until(Object(WebDriverExpectedCondition))
```

I'm assuming it has something to do with the switching to Firefox standalone server. Is there something I need to modify in terms of how **WebDriverExpectedCondition** is used?

I'm using the docker image: `selenium/node-firefox:3.141.59-20210311`

**Is it possible that xpath has to be in a different format for FireFox?**

**Comment by ildarius** [comment]
if anyone runs into this, the issue is not with the `WebDriverWait`, but with the way Xpath needs to be formatted

Chrome and FF differ in how you have to write it, so if you change browsers you may have to update Xpaths as well

**Reply by OndraM** [reply]
Hi @ildarius , nice finding! Do you have a concrete example how did you specify the xpath before and after? It may help others who run into this!

**Reply by ildarius** [reply]
Hi! 

Im not 100% sure how the initial xpath was found for Chrome because I inherited the project, but for firefox the easiest was to right click on the element > inspect > copy > xpath as per below

![screenshot](https://user-images.githubusercontent.com/1866104/115650809-f8d0fe00-a2f7-11eb-9c63-69f0c53dc4f9.png)

I saw someone mention that Firefox has absolute xpath and cannot handle chrome's xpath.
